src/psyclone/psyad/tl2ad.py

```diff
@@ -94,6 +95,13 @@ def generate_adjoint_str(tl_fortran_str, active_variables, create_test=False):
     # to the logger.
     # logger.debug(tl_psyir.view())
 
+    # Apply any required transformations to the TL PSyIR
+    kern_trans(tl_psyir)
```

Comment:
Originally I was thinking that this could be a PSyclone kernel script interface (similar to `trans()` for the PSy layer), that we might add in PSyclone scripts as well - hence the generic name. It would then allow us to simply change to providing a generic user script interface in PSyAD rather than the current hidden one. However, I'm not sure if this is the way we would want to go and it is not difficult to change in the future so I don't really mind. As for an appropriate name, I don't like either of yours :-). I'll go with `preprocess_trans` and see what you think.

Reply:
I like it.